lib/rbac/filterer.rb

```diff
@@ -424,6 +424,10 @@ def scope_to_tenant(scope, user, miq_group)
       user_or_group = user || miq_group
       tenant_id_clause = klass.tenant_id_clause(user_or_group)
 
+      if klass.respond_to?(:tenant_joins_clause)
```

Comment:
Good question. I don't think they should be exclusive. I'm just modifying the current flow of adding additional scopes for tenants. Currently, the scope_to_tenant adds  where clauses. My change adds a new step to the flow by adding additional joins if one is desired. 

Both TenancyMixin and CloudTenancyMixin adds tenant_id_clauses. So :scope_by_cloud_tenant? can't just add the joins_clause. It needs to add the joins_clause and the tenant_id_clause which is currently done in scope_to_tenant. 

I understand that you would like to see a clearer separation by using :scope_by_tenant and :scope_by_cloud_tenant. But it is not clear to me if that is necessary here or how to achieve that cleanly without duplicating code.